comp/snmptraps/status/statusimpl/mock.go

```diff
@@ -41,3 +53,13 @@ func (s *mockManager) GetTrapsPacketsAuthErrors() int64 {
 	defer s.lock.Unlock()
 	return s.trapsPacketsAuthErrors
 }
+
+func (s *mockManager) SetStartError(err error) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.err = err
+}
+
+func (s *mockManager) GetStartError() error {
+	return s.err
```

Comment:
You have to protect `s.err` with lock to not have a race condition.
For example the following code has a race condition:
```
func TestRace1(t *testing.T) {
	var err error
	go func() {
		err = errors.New("test")
	}()
	fmt.Println(err)
}
```

Reply:
Fixed.